rust/relay/src/server/channel_data.rs

```diff
@@ -25,21 +25,22 @@ impl<'a> ChannelData<'a> {
         }
 
         let length = u16::from_be_bytes([data[2], data[3]]);
+        let full_msg_length = 4usize + length as usize;
 
         let actual_payload_length = data.len() - 4;
 
-        if actual_payload_length != length as usize {
+        if data.len() < full_msg_length {
```

Comment:
I think it would be cleaner if we sliced off a `payload` variable from `data` using `data[4..]`.

Then we can conveniently compare `payload.len()` and `length` using `<`.

Finally, we can return `payload[..length]`. This way, we only mention the magic `4` once and everything else is automatically relative.

Thoughts?

Reply:
sounds much cleaner, will update.